### PR TITLE
[DPE-3256] Upgrade test

### PR DESCRIPTION
**Pull request by dragomirp**
Replace the rudimentary upgrade test of the charm upgrading over itself with a copy of the more comprehensive tests in the PG charm

### CODE REVIEW

File: tests/integration/test_upgrade.py [hunk 1]
```diff
@@ -1,100 +1,206 @@
...
+    logger.info("Ensure continuous_writes while in failure state on remaining units")
+    await are_writes_increasing(ops_test)
+
+    logger.info("Re-run pre-upgrade-check action")
+    action = await leader_unit.run_action("pre-upgrade-check")
     await action.wait()
 
-    await ops_test.model.wait_for_idle(apps=APP_NAMES, status="active", raise_on_blocked=True)
+    logger.info("Re-refresh the charm")
+    await application.refresh(path=pgb_charm_jammy)
```
Comment (dragomirp) [inline tests/integration/test_upgrade.py:172]: PG charm tests waited for `waiting` status here, but that seems to fail on juju 3.

File: tests/integration/helpers/ha_helpers.py
Comment (dragomirp) [inline tests/integration/helpers/ha_helpers.py:1]: Mostly a copy paste from the PG VM charm.

File: tests/integration/test_upgrade.py (hunk 1 above)
Comment (taurus-forever) [reply tests/integration/test_upgrade.py:172]: Confirmed: https://bugs.launchpad.net/juju/+bug/2053242

**Review by marceloneppel — APPROVED** [review]
Excellent! LGTM!

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -18,3 +22,16 @@ async def pgb_charm_focal(ops_test: OpsTest):
 async def pgb_charm_jammy(ops_test: OpsTest):
     """Build the pgbouncer charm."""
     return await ops_test.build_charm(".", bases_index=1)
+
+
+@pytest.fixture()
+async def continuous_writes(ops_test: OpsTest) -> None:
+    """Deploy the charm that makes continuous writes to PostgreSQL."""
```
[the remark this answers is not in the document]
Comment (dragomirp) [reply tests/integration/conftest.py]: Updated.

**Pull request merged**